lib/engine/game/g_1812/step/buy_train.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../g_1867/step/buy_train'
+require_relative '../../../step/automatic_loan'
+
+module Engine
+  module Game
+    module G1812
+      module Step
+        class BuyTrain < G1867::Step::BuyTrain
+          include Engine::Step::AutomaticLoan
+
+          def pass!
+            Engine::Step::BuyTrain.instance_method(:pass!).bind_call(self)
+
+            @game.trainless_penalty(current_entity) if current_entity.trains.empty?
+          end
+
+          def discountable_trains_allowed?(_entity)
+            @game.phase.name.to_i >= 5
+          end
+
+          def buy_train_action(action, _entity = nil)
+            @depot_train = action.train.from_depot?
+
+            Engine::Step::BuyTrain.instance_method(:buy_train_action).bind_call(self)
+          end
```

Comment:
You could get rid of this method entirely, by having a `post_train_buy` method in the game class that does nothing (but with a comment explaining why you've done this).

Reply:
done